protos/lobby.proto

```diff
@@ -0,0 +1,100 @@
+import "game_info.proto"
+
+package papan;
+
+message RegisterUpdate {
+  oneof update {
+    GameType type_added = 1;
+    /* We assume that a server will not concurrently add two game types of the
+       same name. If one attempts to do so, the second one should be ignored */
+    string type_removed = 2;
+  }
+}
+
+message RegisterServerResponse {
+}
+
+message CreateGameLobby {
+  uint32 game_type_id = 1;
+  repeated map<string, string> options = 2;
+}
+
+message LobbyAction {
+  CreateGameLobby create = 1;
+}
+
+message GameInstance {
+  uint32 game_type_id = 1;
+  uint32 current_players = 2;
+  uint32 game_lobby_id = 3;
+  map<string, string> chosen_options = 4;
+}
+
+message LobbyUpdate {
+  oneof update {
+    // A server registered a new game types.
+    GameType type_added = 1;
+    // A server unregistered a game type, or disconnected
+    uint32 type_id_removed = 2;
+    // A player created a new game lobby
+    GameInstance instance_added = 1;
+    // A game lobby became unavailable
+    uint32 instance_removed = 2;
+    // The player making this call created a game lobby
+    GameInstance instace_created = 3;
+  }
+}
+
+enum GameLobbyFinish {
+  // Can only be chosen by the game host
+  START = 0;
+  LEAVE = 1;
+}
+
+message GameLobbyAction {
+  oneof action {
+    string send_chat_message = 1;
+    // A finish message should be followed by a client-side half-close
```

Comment:
We'd need the ready set flag from the players present. They should be able to set it at will. 

Also I realize that there are local options for players, such as the color of the token that they play, etc. To make things worse, some of these options may be mutually exclusive. 

We need a method to describe player options in the rest of the system.